### PR TITLE
feat: serverSide validation error markers

### PR DESCRIPTION
**Pull request by roshan-gh**

### CODE REVIEW

File: packages/otelbin/src/components/monaco-editor/otelCollectorConfigValidation.ts [hunk 1]
```diff
@@ -99,6 +102,28 @@ export function validateOtelCollectorConfigurationAndSetMarkers(
 	}
 	if (!totalErrors.jsYamlError) {
 		customValidate(mainItemsData, serviceItemsData, errorMarkers, totalErrors, configData);
+		const serverSideErrorElement = findErrorElement(serverSideValidationPath, parsedYamlConfig);
```
Comment (bripkens) [inline packages/otelbin/src/components/monaco-editor/otelCollectorConfigValidation.ts:105]: **Suggestion:** Add tests for this :)

Comment (roshan-gh) [reply packages/otelbin/src/components/monaco-editor/otelCollectorConfigValidation.ts:105]: All the functions that used for serverSide markers are already have tests. (used the same functions as for ajv error markers) @bripkens

**Comment by roshan-gh** [comment]
@bripkens removed server-side word

File: packages/otelbin/src/components/monaco-editor/otelCollectorConfigValidation.ts (hunk 1 above)
Comment (bripkens) [reply packages/otelbin/src/components/monaco-editor/otelCollectorConfigValidation.ts:105]: So if that is the case, does that mean that this new code can be deleted and there will be a red test? :)

**Comment by mmanciop** [comment]
We need to deduplicate errors on the same path: if two errors are found on the same path, the server-side error takes precedence, see attached screenshot.

Screenshot 2023-11-10 at 11 42 58

**Review by mmanciop — CHANGES_REQUESTED** [review]
We need error deduplication, see comment.

**Review by bripkens — APPROVED** [review]
Let's get this merged and then iterate on the presentation and de-duplication.

**Pull request merged**